compatibility_lib/compatibility_lib/dependency_highlighter.py

```diff
@@ -0,0 +1,267 @@
+
+        if ALLOWED_MINOR_DIFF <= latest['minor'] - install['minor']:
+            msg = ('this dependency is 3 or more minor versions '
+                   'behind the latest version')
+            return Priority(PriorityLevel.HIGH, msg)
+
+        if DEFAULT_GRACE_PERIOD_IN_DAYS < elapsed_time.days:
+            msg = ('it has been over 6 months since the latest version '
+                   'for this dependency was released')
+            return Priority(PriorityLevel.HIGH, msg)
+
+        msg = 'this dependency is not up to date with the latest version'
+        return Priority(PriorityLevel.LOW, msg)
+
+    def check_package(self, package_name):
+        """Looks for and returns outdated dependencies for a single package
+
+        Args:
+            package_name: the name of the package to query (string)
+        Returns:
+            a list of outdated dependencies
+        """
+        dependency_info = self._get_dependency_info(package_name)
+        outdated_dependencies = []
+        for name, info in dependency_info.items():
+            priority = Priority()
+            try:
+                install = _sanitize_release_tag(info['installed_version'])
+            except UnstableReleaseError as err:
+                priority = Priority(PriorityLevel.HIGH, str(err))
+
+            if not info['is_latest'] or priority.level != PriorityLevel.NOT_SET:
+                latest =  _sanitize_release_tag(info['latest_version'])
+                elapsed_time = info['current_time']-info['latest_version_time']
+
+                if priority.level == PriorityLevel.NOT_SET:
+                    priority = self._get_update_priority(
+                        install, latest, elapsed_time)
+                dependency = OutdatedDependency(
+                    name, package_name, priority, info)
+                outdated_dependencies.append(dependency)
+        return outdated_dependencies
+
+    def check_packages(self, packages, max_workers=20):
+        """Looks for and returns outdated dependencies for multiple package
+
+        Args:
+            packages: a list of package names to query (string)
+        Returns:
+            a dict mapping dependency name to outdated dependencies
+        """
+        results = {}
+        with concurrent.futures.ThreadPoolExecutor(
+                max_workers=max_workers) as p:
+            outdated_dependencies = p.map(self.check_package, packages)
+
+            for pkgname, result in zip(packages, outdated_dependencies):
+                results[pkgname] = result
+        return results
+
+
+def _sanitize_release_tag(release):
+    """Throws an error if the given release version is unstable
+
+    Args:
+        release: the semantic release as a string
+    Returns:
+        a dict that maps the major, minor, and patch (represented as strings)
+        to the int value of those fields
+    """
+    patch_match = r'^\d+\.\d+$'
+    match = r'^\d+\.\d+\.\d+$|^\d+\.\d+\.\d+\.\d+$'
```

Comment:
An example of unstable release version number will be helpful.

Reply:
Is this ok?
https://github.com/GoogleCloudPlatform/cloud-opensource-python/blob/d115e80ebf8b49f06ef369480814c847dabc7702/compatibility_lib/compatibility_lib/dependency_highlighter.py#L232